appium/lib/screen-object/accessors/element.rb

```diff
@@ -81,44 +81,119 @@ def dynamic_text_exists? dynamic_text
         end
       end
 
-      def scroll
-        $driver.execute_script 'mobile: scrollTo',:element => element.ref
-        # $driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
+      def gesture(x1,y1,x2,y2,duration)
+        Appium::TouchAction.new($driver).swipe(start_x: x1, start_y: y1, end_x: x2, end_y: y2,duration: duration).perform
+      rescue
+        raise("Error during gesture")
       end
 
-      def scroll_to_text(text)
-        $driver.scroll_to(text)
+      def _scroll(direction = :down, duration = 1000)
+        size = driver.window_size
+        x = size.width/2
+        y = size.height/2
+        case direction
+        when :up then loc = [x, y*0.5, x, (y + (y*0.5)),duration]
+        when :down then loc =[x, y, x, y * 0.5,duration]
+        when :left then loc = [x * 0.6, y, x * 0.3, y,duration]
+        when :right then loc =[x * 0.3, y, x * 0.6, y,duration]
+        else
+          raise('Only upwards and downwards scrolling are supported')
+        end
+        gesture(loc[0],loc[1],loc[2],loc[3],loc[4])
       end
 
-      def scroll_to_exact_text(text)
-        $driver.scroll_to_exact(text)
+      def scroll_down
+        _scroll(:down)
       end
 
-      def scroll_for_element_click
-        if element.displayed?
-          element.click
+      def scroll_up
+        _scroll(:up)
+      end
+
+      def swipe_left
+        _scroll(:left)
+      end
+
+      def swipe_right
+        _scroll(:right)
+      end
+
+      def swipe_my_element(direction = :down, duration = 1000)
+        my_element = element.rect
+        start_x = my_element.x
+        end_x = my_element.x + my_element.width
+        start_y = my_element.y
+        end_y = my_element.y + my_element.height
+        height = my_element.height
+        case direction
+        when :up then  loc = [end_x * 0.5, (start_y + ( height * 0.2)), end_x * 0.5, (end_y - (height * 0.2)),duration]
+        when :down then  loc = [end_x * 0.5, (end_y - (height * 0.2)), start_x * 0.5, (start_y + (height * 0.3)), duration]
+        when :left then  loc = [end_x * 0.9,  end_y - (height/2), start_x, end_y - (height/2), duration]
+        when :right then  loc = [end_x * 0.1, end_y - (height/2), end_x * 0.9, end_y - (height/2), duration]
         else
-          scroll
-          element.click
+          raise('Only upwards and downwards scrolling are supported')
         end
+        gesture(loc[0],loc[1],loc[2],loc[3],loc[4])
```

Comment:
Same as mentioned above. The `swipe` method can be undeleted and the gesture call can go there.